app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
OK lah, let's pray for the new guys :-)

On Thu, Jan 8, 2015 at 6:21 PM, Joel Low <notifications@github.com> wrote:

> In app/models/instance.rb
> <https://github.com/Coursemology/coursemology2/pull/41#discussion-diff-22644170>
> :
>
> > @@ -15,6 +15,9 @@ def find_tenant_by_host(host)
> >      end
> >    end
> >
> > +  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
> > +  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
>
> not always possible, have you seen alias_method_chain? or instance_eval
> or class_eval used with strings (yes...)
>
> ah, the powers of Ruby metaprogramming.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Coursemology/coursemology2/pull/41/files#r22644170>.
>